### ISSUE TITLE
My blog doesn't build with current master

### ISSUE THREAD

**Issue by ifraixedes** [spf13/hugo#1879]
I don't know if this helpful or not but it may help to spot what is going on.

I tried to build my [blog](https://github.com/ifraixedes/blog) with the current master,  the process blocked at some point and never finished and logs didn't show any error or warning message only the INFO ones.

I tried to go down several commits and reach one that I know that it had been working but it didn't either finish.

Because I did all this process with the new go release 1.6, I wanted to see if that also happened with 1.5.3 and I've realised that current master, with Go 1.5.3, finishes the process without any issue.

Do you have any idea what is happening?

**Comment by bep** [comment]
This fits better on https://discuss.gohugo.io/

**Comment by bep** [comment]
OK, I notice you mention "process being blocked" ... Reopen.

**Comment by tatsushid** [comment]
It is reproduced by using @ifraixedes blog source on my machine and found where it was blocked. https://github.com/spf13/hugo/blob/master/hugolib/site.go#L1372 is there. It sends an error via a channel but it is not listened by anywhere.

I forgot the related issue but in Go 1.6, template is treated more strictly and it causes a template error. I saw a following error in this investigation.

```
&errors.errorString{s:"html/template: cannot redefine \"_internal/shortcodes/ref.html\" after it has executed"}
```

The code itself could hit this problem but in Go 1.5 or earlier, it is luckily avoided.

**Comment by tatsushid** [comment]
The blocking issue is resolved in commit e985364 but it stops rendering the site with a template error. Need more investigation.

**Comment by bep** [comment]
It's the `/content/rss/index.html' that makes this fail. I have questioned this logic before. I guess the solution is to create a clone of s.Tmpl before any execution, then use that in:

https://github.com/spf13/hugo/blob/master/hugolib/site.go#L1372

**Comment by tatsushid** [comment]
@bep, I agree with you. This is the issue discussed at #1470. And one more, shortcode templates are rendered before calling `RenderPages()` so any redefinition at https://github.com/spf13/hugo/blob/master/hugolib/site.go#L1372 may violate the redefinition rule introduced in golang/go@12dfc3b.

As you mentioned, one of the solution is creating clone before executing templates or if the `IsRenderable` check block is really unneeded today, just removing this also seems to be OK.

**Comment by bep** [comment]
The IsRenderable is used by some (the blog in this issue is one), so a clone may be simplest.

**Comment by ifraixedes** [comment]
I'm sorry but I don't have that much knowledge of the hugo code base to fully understand your discussion, so I cannot to contribute, so far, that much in fixes.

After [e985364 ](https://github.com/spf13/hugo/commit/e9853642e5214d66357d2622978f0c85019eb24c)  as @tatsushid the blocking issues is solved, but the site still not rendering.

I've been investigating which was the problem and I found that it was complaining about one of the shortcuts that my theme has, whose name (youtube) collides with one provided by hugo.

I renamed it, but I'm still not able to render the site, because hugo collides with a default template which I override in the theme (`themes/fraixedes/layouts/_default/list.html`).

The error log message was:
`Error building site: Error(s) rendering pages: html/template: cannot redefine "theme/_default/list.html" after it has executed`
